frontend/app/ui_components/date/date-range-picker-directive.js

```diff
@@ -0,0 +1,183 @@
+
+              if(buttonPane.find('.ui-datepicker-clear').length == 0) {
+                buttonPane.find('.ui-datepicker-clear').remove();
+              }
+
+              jQuery( '<button>', {
+                    text: 'Clear',
+                    click: function() {
+                      inp.val('');
+                      inp.change();
+                    }
+                })
+                .addClass('ui-datepicker-clear ui-state-default ui-priority-primary ui-corner-all')
+                .appendTo(buttonPane);
+            }, 1);
+          },
+          setDate = function(input, date, noDateText) {
+            console.log(date);
+            if(date) {
+              input.datepicker('option', 'defaultDate', formattedDate(date));
+              input.datepicker('option', 'setDate', formattedDate(date));
+              input.val(formattedDate(date))
+            } else {
+              input.datepicker('option', 'defaultDate', null);
+              input.datepicker('option', 'setDate', null);
+              input.val(noDateText);
+              date = null;
+            }
+            console.log(input.val());
+          };
+
+      inputStart.on('change', function() {
+        if(inputStart.val().replace(/^\s+|\s+$/g, '') === '') {
+          $timeout(function() {
+            scope.startDate = null;
+          });
+          inputStart.val(noStartDate);
+          $timeout.cancel(startTimerId);  
+          return;
+        }
+        $timeout.cancel(startTimerId);
+        startTimerId = $timeout(function() {
+          var date = inputStart.val(),
+              isValid = TimezoneService.isValid(date, 'DD/MM/YYYY');
+
+          if(isValid){
+            scope.startDate = formattedISODate(date);
+          }
+        }, 1000)
+      });
+
+      inputEnd.on('change', function() {
+        if(inputEnd.val().replace(/^\s+|\s+$/g, '') === '') {
+          $timeout(function() {
+            scope.endDate = null;
+          });
+          inputEnd.val(noEndDate);
+          $timeout.cancel(endTimerId);  
+          return;
+        }
+        $timeout.cancel(startTimerId);
+        startTimerId = $timeout(function() {
+          var date = inputEnd.val(),
+              isValid = TimezoneService.isValid(date, 'DD/MM/YYYY');
+
+          if(isValid){
+            scope.endDate = formattedISODate(date);
+          }
+        }, 1000)
+      });
+      setDate(inputStart, scope.startDate, noStartDate);
+      setDate(inputEnd, scope.endDate, noEndDate);
+      inputStart.datepicker({
+        firstDay: ConfigurationService.startOfWeek(),
+        showWeeks: true,
+        changeMonth: true,
+        numberOfMonths: 2,
+        showButtonPanel: true,
+        dateFormat: 'dd/mm/yy',
+        beforeShow: function(inp) {
+          addClearButton(inputStart, noStartDate, scope.startDate, scope.editableStartDate);
+        },
+        onChangeMonthYear: function(year, month, inst) {
```

Comment:
'inst' is defined but never used.<br>'month' is defined but never used.<br>'year' is defined but never used.